addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/handler/AirVisualNodeHandler.java

```diff
@@ -0,0 +1,267 @@
+public class AirVisualNodeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AirVisualNodeHandler.class);
+
+    public static final String NODE_JSON_FILE = "latest_config_measurements.json";
+
+    private final Gson gson;
+
+    private ScheduledFuture<?> pollFuture;
+
+    private long refreshInterval;
+
+    private String nodeAddress;
+
+    private String nodeUsername;
+
+    private String nodePassword;
+
+    private String nodeShareName;
+
+    private NodeData nodeData;
+
+    public AirVisualNodeHandler(@NonNull Thing thing) {
+        super(thing);
+        gson = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing AirVisual Node handler");
+
+        AirVisualNodeConfig config = getConfigAs(AirVisualNodeConfig.class);
+
+        if (config.address == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Node address must be set");
+            return;
+        }
+        this.nodeAddress = config.address;
+
+        this.nodeUsername = config.username;
+
+        if (config.password == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Node password must be set");
+            return;
+        }
+        this.nodePassword = config.password;
+
+        this.nodeShareName = config.share;
+
+        this.refreshInterval = config.refresh * 1000L;
+
+        schedulePoll();
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Node poll scheduled");
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId(), true);
+        } else {
+            logger.debug("Can not handle command '{}'", command);
+        }
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
+        stopPoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private synchronized void stopPoll() {
+        if (pollFuture != null && !pollFuture.isCancelled()) {
+            pollFuture.cancel(false);
+        }
+    }
+
+    private synchronized void schedulePoll() {
+        logger.debug("Scheduling poll for 500ms out, then every {} ms", refreshInterval);
+        pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 500, refreshInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollNode();
+            updateStatus(ThingStatus.ONLINE);
+        } catch (IOException e) {
+            logger.debug("Could not connect to Node", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (Exception e) {
```

Comment:
Please do not catch `Exception`, catch a more specific one instead. Because otherwise you might accidentally also catch `RuntimeExceptions` like `NullpointerExceptions`, etc.